durablestore/dynamodb/dynamodb.go

```diff
@@ -40,14 +40,14 @@ func (ddb ddb) GetItem(ctx context.Context, persistenceID string) (*item, error)
 		TableName: aws.String(ddb.tableName),
 		Key:       key,
 	})
-	if err != nil {
```

Comment:
I suggest checking the error before the `result.Item` code path.

Reply:
That makes sense. I improved the way we handle the cases here now